File: app/src/main/java/io/github/fvasco/pinpoi/util/DismissOnClickListener.java
package io.github.fvasco.pinpoi.util;

import android.content.DialogInterface;

/**
 * Dismiss dialog
 *
 * @author Francesco Vasco
 */
public enum DismissOnClickListener implements DialogInterface.OnClickListener {
    INSTANCE;

    @Override
    public void onClick(DialogInterface dialog, int which) {
        dialog.dismiss();
    }
}

File: app/src/main/java/io/github/fvasco/pinpoi/util/DismissOnClickListener.kt
package io.github.fvasco.pinpoi.util

import android.content.DialogInterface

/**
 * Dismiss dialog

 * @author Francesco Vasco
 */
object DismissOnClickListener : DialogInterface.OnClickListener {

    override fun onClick(dialog: DialogInterface, which: Int) {
        dialog.dismiss()
    }
}
